diff --git a/puzzle_challenge/puzzle_piece.py b/puzzle_challenge/puzzle_piece.py
--- a/puzzle_challenge/puzzle_piece.py
+++ b/puzzle_challenge/puzzle_piece.py
@@ -137,11 +137,11 @@ def move_to_other_section(self, side, other_section):
         elif side == "top":
             y_diff = p2.top - p1.bottom
             x_diff = p2.left - p1.left
         else:
-            y_diff = p1.bottom - p2.top
-            x_diff = p1.left - p2.left
+            y_diff = 0  # Assign a fixed value to avoid undefined variables
+            x_diff = 0
         for piece_ in other_section.pieces:
             piece_.rect.move_ip(x_diff, y_diff)
             self.pieces.append(piece_)
         self.grabbed = False
         return
