diff --git a/labels.py b/labels.py
--- a/labels.py
+++ b/labels.py
@@ -133,7 +133,9 @@ class Label:
         self.image = self.idle_image
         self.clicked = False
         self.hover = False
-    def render_text(self):
+    def render_text(self):
         font, size = self.font, self.font_size
         if (font, size) not in LOADED_FONTS:
             LOADED_FONTS[font, size] = pg.font.Font(font, size)
+        # Fixing potential issues
+        x_diff, y_diff = self.calculate_diff()
+        if x_diff is not None and y_diff is not None:
+            self.x_diff = x_diff
+            self.y_diff = y_diff