diff --git a/puzzle-challenge/puzzle_piece.py b/puzzle-challenge/puzzle_piece.py
--- a/puzzle-challenge/puzzle_piece.py
+++ b/puzzle_piece.py
@@ -134,7 +134,9 @@ class PuzzlePiece:
         elif side == "bottom":
             y_diff = p1.bottom - p2.top
             x_diff = p1.left - p2.left
-        for piece_ in other_section.pieces:
+        x_diff = p1.left - p2.left  # Move this line up to fix 'possibly-used-before-assignment'
+        y_diff = p1.bottom - p2.top
         piece_.rect.move_ip(x_diff, y_diff)
         self.pieces.append(piece_)
     def get_event(self, event):
         self.grabbed = False
         return